components/renku_data_services/namespace/db.py

```diff
@@ -74,14 +74,20 @@ async def generate_user_namespaces(self, *, session: AsyncSession | None = None)
         return output
 
     async def get_groups(
-        self,
-        user: base_models.APIUser,
-        pagination: PaginationRequest,
+        self, user: base_models.APIUser, pagination: PaginationRequest, direct_member: bool = False
     ) -> tuple[list[models.Group], int]:
         """Get all groups from the database."""
+        if direct_member:
+            group_ids = await self.authz.resources_with_direct_membership(user, ResourceType.group)
+
         async with self.session_maker() as session, session.begin():
-            stmt = select(schemas.GroupORM).limit(pagination.per_page).offset(pagination.offset)
-            stmt = stmt.order_by(schemas.GroupORM.creation_date.asc(), schemas.GroupORM.id.asc(), schemas.GroupORM.name)
+            stmt = select(schemas.GroupORM)
+            if direct_member:
+                stmt = stmt.where(schemas.GroupORM.id.in_(group_ids))
+            stmt = stmt.limit(pagination.per_page).offset(pagination.offset)
+            stmt = stmt.order_by(
+                schemas.GroupORM.creation_date.desc(), schemas.GroupORM.id.desc(), schemas.GroupORM.name
+            )
```

Comment:
You can sort by `id.desc()` here. IDs are unique and sorting them will correspond to the time the ID was created.

```suggestion
            stmt = stmt.order_by(schemas.GroupORM.id.desc())
```